homeassistant/components/panasonic_viera/config_flow.py

```diff
@@ -0,0 +1,165 @@
+            }
+
+            self._data = {**self._data, **encryption_data}
+
+            return self.async_create_entry(
+                title=self._data[CONF_NAME], data=self._data,
+            )
+
+        return self._show_pair_form()
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        await self.async_load_data(import_config)
+        return await self.async_create_remote()
+
+    async def async_load_data(self, config):
+        """Load the data."""
+        self._data = config
+
+        self._data[CONF_PORT] = (
+            self._data[CONF_PORT] if CONF_PORT in self._data else DEFAULT_PORT
+        )
+        self._data[CONF_ON_ACTION] = (
+            self._data[CONF_ON_ACTION] if CONF_ON_ACTION in self._data else None
+        )
+
+        await self.async_set_unique_id(self._data[CONF_HOST])
+        self._abort_if_unique_id_configured()
+
+    async def async_create_remote(self):
+        """Create the remote."""
+        try:
+            self._remote = await self.hass.async_add_executor_job(
+                partial(RemoteControl, self._data[CONF_HOST], self._data[CONF_PORT])
+            )
+        except (TimeoutError, URLError, OSError) as err:
+            _LOGGER.error("Could not establish remote connection: %s", err)
+            self._errors = {"base": ERROR_NOT_CONNECTED}
+            return await self.async_step_user()
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.error("An unknown error occurred: %s", err)
+            self._errors = {"base": ERROR_UNKNOWN}
+            return await self.async_step_user()
+
+        if self._remote.type == TV_TYPE_ENCRYPTED:
+            return await self.async_step_pairing()
+
+        return self.async_create_entry(title=self._data[CONF_NAME], data=self._data,)
+
+    @callback
+    def _show_user_form(self):
+        """Show the initial form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_HOST,
+                        default=self._data[CONF_HOST]
+                        if self._data[CONF_HOST] is not None
+                        else "",
+                    ): str,
+                    vol.Optional(
+                        CONF_NAME,
+                        default=self._data[CONF_NAME]
+                        if self._data[CONF_NAME] is not None
+                        else DEFAULT_NAME,
+                    ): str,
+                }
+            ),
+            errors=self._errors,
+        )
+
+    @callback
+    def _show_pair_form(self):
+        """Show the pairing form to the user."""
+        if self._errors is not None and self._errors["base"] in [
+            ERROR_NOT_CONNECTED,
+            ERROR_UNKNOWN,
+        ]:
+            self._errors = None
```

Comment:
Why do reset the errors in these cases?

Reply:
The only error that can appear during the pairing step is the invalid PIN code one, so, in case there are remaining errors from previous steps still stored, they are wiped out and not shown to the user